### ISSUE TITLE
Transition frequent itemset mining algorithms to use Boolean dataframes

### ISSUE THREAD

**Issue by rasbt** [rasbt/mlxtend#620]
As noted in #619, apriori is substantially faster on Boolean types data arrays for certain settings.

It would be good to add a warning about non Boolean inputs and encourage users to use Boolean DataFrames (if we deprecate integer dataframes, we can also simplify the code)

In a similar manner, it'd be good to check how the input dataframe type affects

- `fpgrowth`
- `fpmax`

**Comment by NimaSarajpoor** [comment]
@rasbt 
I think I used this package about two years ago for building a simple recommendation model. To rebuild my memory and start contributing to this package, I intentionally tried to find an "easy" issue. So, if you do not mind, I can start working on this issue for now. Do you have any suggestion?

**Comment by rasbt** [comment]
Cool, thanks for the contributions! Btw the stat479 was something I tagged for students in my class who wanted to contribute to open source projects, please ignore!

**Comment by rasbt** [comment]
I think that 0/1 (or non-Boolean) types are currently still supported. But the goal would be to show a DeprecationWarning if someone uses them.

**Comment by NimaSarajpoor** [comment]
@rasbt 
Ahh....got it! I was thinking of 0/1 as Boolean values. My bad.

**Comment by NimaSarajpoor** [comment]
@rasbt 
Sorry again for my lame (?) question. I changed the `apriori.py` file. Then, I tried to reinstall the package with `python setup.py install` to make the changes effective, so I can try them out on my end. However, that does not show my changes when I want to call `apriori.py`. In fact, I added a new parameter `is_bool_dtype=False` and did `git add` and `git commit`...and then reinstall the package (this is what I do for developing STUMPY package)...but then I cannot see the new parameter.

I also took a look at the `How to Contribute` but couldn't find any clue there. I do not want to write a unit test to check the warning (or should I ?!). I wanted to just use a simple notebook to see if I get a warning or not.

Would you please let me know what I have been missing so far?

**Comment by rasbt** [comment]
Hm, this is strange. Did you perhaps change the git branch after making the changes and your changes are on a different branch?

Btw. I would recommend using 

    python setup.py develop

over

    python setup.py install

which will update the package when you have changes.

**Comment by NimaSarajpoor** [comment]
No, I did not change the branch or environment. I created a new environment, and there, I created a dev branch.... and I am doing all of my work there (and then I will push my dev branch to upstream). 

I tried `python setup.py develop`. did not work...maybe I messed up with something...I am going to do the whole process again starting with forking the project one more time and clone it and see how things go. Will update you.
